src/embezzler.ts

```diff
@@ -169,6 +180,127 @@ export const embezzlerSources = [
+      cliExecuteOutput("timespinner list monsters").includes(
+        $monster`Knob Goblin Embezzler`.name
+      ) &&
+      get("_timeSpinnerMinutesUsed") <= 7,
+    () =>
+      have($item`Time-Spinner`) &&
+      cliExecuteOutput("timespinner list monsters").includes($monster`Knob Goblin Embezzler`.name)
+        ? Math.min((10 - get("_timeSpinnerMinutesUsed")) / 3)
+        : 0,
+    (options: EmbezzlerFightOptions) => {
+      const macro = options.macro ?? embezzlerMacro();
+      withMacro(macro, () => {
+        visitUrl(`inv_use.php?whichitem=${toInt($item`Time-Spinner`)}`);
+        runChoice(1);
+        visitUrl(
+          `choice.php?whichchoice=1196&monid=${$monster`Knob Goblin Embezzler`.id}&option=1`
+        );
+      });
+    }
+  ),
+  new EmbezzlerFight(
+    "Macrometeorite",
+    () =>
+      get("beGregariousMonster") === $monster`Knob Goblin Embezzler` &&
+      get("beGregariousFightsLeft") > 0 &&
+      have($skill`Meteor Lore`) &&
+      get("_macrometeoriteUses") < 10,
+    () =>
+      get("beGregariousMonster") === $monster`Knob Goblin Embezzler` &&
+      get("beGregariousFightsLeft") > 0 &&
+      have($skill`Meteor Lore`)
+        ? 10 - get("_macrometeoriteUses")
+        : 0,
+    (options: EmbezzlerFightOptions) => {
+      saberCrateIfDesired();
+      if (
+        have($item`miniature crystal ball`) &&
+        get("crystalBallLocation") === $location`Noob Cave` &&
+        get("crystalBallMonster") !== $monster`Knob Goblin Embezzler` &&
+        !(get("_saberForceMonster") === $monster`crate` && get("_saberForceMonsterCount") > 0)
+      ) {
+        equip($slot`familiar`, $item`miniature crystal ball`);
+      }
+      const baseMacro = options.macro ?? embezzlerMacro();
+      const macro = Macro.if_($monster`crate`, Macro.skill($skill`Macrometeorite`)).step(baseMacro);
+      adventureMacro($location`Noob Cave`, macro);
+    }
+    //do we want to equip orb on these guys?
+  ),
+  new EmbezzlerFight(
+    "Powerful Glove",
+    () =>
+      get("beGregariousMonster") === $monster`Knob Goblin Embezzler` &&
+      get("beGregariousFightsLeft") > 0 &&
+      have($item`Powerful Glove`) &&
+      get("_powerfulGloveBatteryPowerUsed") < 90,
+    () =>
+      get("beGregariousMonster") === $monster`Knob Goblin Embezzler` &&
+      get("beGregariousFightsLeft") > 0 &&
+      have($item`Powerful Glove`)
+        ? Math.min((100 - get("_powerfulGloveBatteryPowerUsed")) / 10)
+        : 0,
+    (options: EmbezzlerFightOptions) => {
+      saberCrateIfDesired();
+      if (
+        have($item`miniature crystal ball`) &&
+        get("crystalBallLocation") === $location`Noob Cave` &&
+        get("crystalBallMonster") === $monster`crate`
+      ) {
+        equip($slot`familiar`, $item`miniature crystal ball`);
+      }
+      const baseMacro = options.macro ?? embezzlerMacro();
+      const macro = Macro.if_($monster`crate`, Macro.skill($skill`CHEAT CODE: Replace Enemy`)).step(
```

Comment:
Theoretically, do we want to extract etc. here before macro-ing or are we too afraid of winning the fight from random effects to risk the extra 10 meat?

Reply:
I am far too afraid to lose the fight from elemental-fairy-in-the-bjorn damage. We can do stuff with the existing stasis code but I'm declaring that out of scope of this PR.

Reply:
That makes complete sense and I totally support that (and agree that it is likely out of scope). I just wanted to make sure that this was something that was considered and rejected.
I thought that not beating a crate was easy, but based on some testing I did while overdrunk I was able to kill crates with 100ML too quickly with just regular attacks. And losing a turn here is p bad.